packages/data-stores/src/onboard/utils.ts

```diff
@@ -28,6 +28,11 @@ export const goalsToIntent = ( goals: SiteGoal[] ): SiteIntent => {
 		return SiteIntent.Write;
 	}
 
+	// Newsletter flow
+	if ( goals.includes( SiteGoal.Newsletter ) ) {
+		return SiteIntent.NewsletterV2;
+	}
```

Comment:
Do you think we should throw a feature flag around this? It'd allow us to merge this PR then give a link to Autumn et al to show it working?